### DISCUSSION TITLE
Shared solar plant

### DISCUSSION THREAD

**Discussion by galodi** [tijsverkoyen/HomeAssistant-FusionSolar#151] [Q&A] [answered]
Hi,

I have a shared solar plant with my neighbors where I get quota 15% of the plant production but the data I get through the API and the integration is 100% of the plant production.

Is there any way this 15% quota can be set in the HA energy dashboard?

Thanks

**Comment by tijsverkoyen** [comment]
Is this sharing a feature in the Huawei Software? Of just an agreement between you and your neighbours?
I don't recall anything about this in the documentation. 

I think the best way for you is to work with helpers that take 15% from the values you want to interact on.

**Comment by galodi** [comment]
Thanks for your feedback. Yes, this is an agreement between neighbor and not feature of the software. Following your tip I am trying to create a new sensor as follows but I unable to get it shown in the solar panels sensor in the energy dashboard. Any suggestion what I am doing wrong?

sensor:
  - platform: template
    sensors:
      solar_production_15_percent:
        unique_id: "Solar_Production_15%"
        friendly_name: "Solar Production (15%)"
        device_class: energy
        value_template: "{{ states('sensor.XXXX_total_current_day_energy') | float * 0.15 }}"
        unit_of_measurement: "kWh"

![image](https://github.com/tijsverkoyen/HomeAssistant-FusionSolar/assets/160169019/a5ad18eb-60bf-4e6e-b817-63341ddd98a3)

**Comment by tijsverkoyen** [comment]
In the energy dashboard you need to use the `sensor.XXX_total_lifetime_energy` sensor

**Comment by tijsverkoyen** [comment]
@galodi did that work?

**Reply by galodi** [reply, marked as answer]
Yes, I finally got it to work but I had to create the custom sensor through the helpers option instead of directly editing the configuration.yaml file. Now it is working fine and I get in the energy dashboard the 15% of the production. 
Many thanks for your help and your integration which works great!
